### PR TITLE
Github Deployments: Make UI tweaks to make it more consistent and clear

### PR DESCRIPTION
**Pull request by vykes-mac**
Closes https://github.com/Automattic/dotcom-forge/issues/5940
Related to 
https://github.com/Automattic/dotcom-forge/issues/5926
https://github.com/Automattic/dotcom-forge/issues/5949
https://github.com/Automattic/dotcom-forge/issues/5947
## Proposed Changes

* Update UI based on design suggestions in the issues

## Testing Instructions

*

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

### CODE REVIEW

File: client/my-sites/github-deployments/deployments/index.tsx [hunk 1]
```diff
@@ -47,6 +51,10 @@ export function GitHubDeployments() {
 	};
 
 	const renderContent = () => {
+		if ( isLoading ) {
```
Comment (zaguiini) [inline client/my-sites/github-deployments/deployments/index.tsx:54]: I think it's fine we show stale data while refetching. What problem were you trying to solve here?

Comment (vykes-mac) [reply client/my-sites/github-deployments/deployments/index.tsx:54]: The UI appear glitchy, there was the loading placeholder , then the connection form flashes before showing the deployment list.

Comment (vykes-mac) [reply client/my-sites/github-deployments/deployments/index.tsx:54]: In other words the `installations` finish loading before the deployments and that caused the connection form to be displayed. Then when the deployments finish loading the deployment list is shown.

Comment (zaguiini) [reply client/my-sites/github-deployments/deployments/index.tsx:54]: I see. That's a weird situation. I think the appropriate would be:

- display placeholder if no data
- load deployments while refreshing then hydate deployments or display connection form or authorization

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/11585388

Some locales (Hebrew, Japanese) have been temporarily machine-translated due to translator availability. All other translations are usually ready within a few days. Untranslated and machine-translated strings will be sent for translation next Monday and are expected to be completed by the following Friday.

Hi @vykes-mac, could you please edit the description of this PR and add a screenshot for our translators? Ideally it'd include all of the following strings:
- `Deployment runs`
- `Deploying`
- `Configure connection`

Thank you in advance!

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/11585388).

**Pull request merged**